contracts/consensus/Coconsensus.sol

```diff
@@ -352,4 +553,140 @@ contract Coconsensus is MasterCopyNonUpgradable, GenesisCoconsensus, MosaicVersi
             observer.setup();
         }
     }
+
+    /**
+     * @notice Assert the opening of kernel in consensus cogateway.
+     *
+     * @param _protocore Protocore contract address.
+     * @param _kernelHeight New kernel height
+     * @param _updatedValidators  The array of addresses of the updated validators.
+     * @param _updatedReputation The array of reputation that corresponds to
+     *                        the updated validators.
+     * @param _gasTarget The gas target for this metablock
+     * @param _transitionHash Transition hash.
+     * @param _source Blockhash of source checkpoint.
+     * @param _target Blockhash of target checkpoint.
+     * @param _sourceBlockNumber Block number of source checkpoint.
+     * @param _targetBlockNumber Block number af target checkpoint.
+     *
+     * \pre Kernel hash generated from given input parameters is equal to the
+     *      kernel hash form consensus cogateway contract for the given
+     *      `_kernelHeight`.
+     *
+     * \post return the kernel hash.
+     */
+    function assertOpenKernel(
+        ProtocoreI _protocore,
```

Comment:
I would propose to pass domainSeparator itself, however it's not crucial to fix.